File: Editorial/171_Excel_Sheet_Column_Number/C/2_Left_to_Right.c
int titleToNumber(char* s) {
    int result = 0;
    int n = strlen(s);
    for (int i = 0; i < n; i++) {
        result = result * 26;
        // In C, subtracting characters is subtracting ASCII values of
        // characters
        result += (s[i] - 'A' + 1);
    }
    return result;
}

File: Editorial/0101-0200/0171_Excel_Sheet_Column_Number/TypeScript/2_Left_to_Right.ts
function titleToNumber(s: string): number {
    let result = 0;
    let n = s.length;
    for (let i = 0; i < n; i++) {
        result = result * 26;
        // In Typescript, subtracting characters is subtracting ASCII values of characters
        result += s.charCodeAt(i) - "A".charCodeAt(0) + 1;
    }
    return result;
}
